packages/server/src/api/controllers/row/alias.ts

```diff
@@ -0,0 +1,125 @@
+import { QueryJson, SearchFilters, Table, Row } from "@budibase/types"
+import { getDatasourceAndQuery } from "../../../sdk/app/rows/utils"
+import { cloneDeep } from "lodash"
+
+export default class AliasTables {
+  character: string
+  aliases: Record<string, string>
+  tableAliases: Record<string, string>
+  tableNames: string[]
+
+  constructor(tableNames: string[]) {
+    this.tableNames = tableNames
+    this.character = "a"
+    this.aliases = {}
+    this.tableAliases = {}
+  }
+
+  getAlias(tableName: string) {
+    if (this.aliases[tableName]) {
+      return this.aliases[tableName]
+    }
+    const char = this.character
+    this.aliases[tableName] = char
+    this.tableAliases[char] = tableName
+    this.character = String.fromCharCode(char.charCodeAt(0) + 1)
```

Comment:
You probably want to do 1 of 2 things here:

1. Put a guard in place so that when you get to `"z"`, an error is thrown. The next character is `"{"` and I'm not sure that'd be valid as an alias.
2. Fall back to 2 character aliases, then 3, etc.